### PR TITLE
Implement methods for native security chain

### PR DESCRIPTION
**Pull request by phiz71**
**Issue**

https://gravitee.atlassian.net/browse/APIM-7143

**Description**

- implements native kafka security methods
- use new HttpSecurityPolicy interface

BREAKING CHANGE: requires APIM 4.6+

---
**Gravitee.io Automatic Deployment**

🚀 A prerelease version of this package has been published on Gravitee's private artifactory, you can:
 - use it directly by updating your project with version: `5.0.0-apim-7143-impl-security-chain-SNAPSHOT`
 - download it from Artifactory [here](https://odbxikk7vo-artifactory.services.clever-cloud.com/gravitee-snapshots/io/gravitee/policy/gravitee-policy-apikey/5.0.0-apim-7143-impl-security-chain-SNAPSHOT/gravitee-policy-apikey-5.0.0-apim-7143-impl-security-chain-SNAPSHOT.zip)

### CODE REVIEW

File: src/main/java/io/gravitee/policy/apikey/ApiKeyPolicy.java [hunk 1]
```diff
@@ -176,11 +182,103 @@ private Optional<String> extractApiKey(HttpExecutionContext ctx) {
...
+                    return Maybe.just(SecurityToken.forMD5ApiKey(md5ApiKey));
+                }
+                return Maybe.just(SecurityToken.invalid(MD5_API_KEY));
+            }
+        }
+        return Maybe.empty();
+    }
+
+    @Override
+    public Completable authenticate(KafkaConnectionContext ctx) {
```
Comment (phiz71) [inline src/main/java/io/gravitee/policy/apikey/ApiKeyPolicy.java:211]: @jhaeyaert should it be executed in a Completable.defer() ?

Comment (jhaeyaert) [reply src/main/java/io/gravitee/policy/apikey/ApiKeyPolicy.java:211]: Yes 👍

**Review by ThibaudAV — APPROVED** [review]

**Pull request merged**